pubsub/cloud-client/subscriber.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to perform basic operations on
+subscriptions with the Cloud Pub/Sub API.
+
+For more information, see the README.md under /pubsub and the documentation
+at https://cloud.google.com/pubsub/docs.
+"""
+
+import argparse
+
+from gcloud import pubsub
+
+
+def list_subscriptions(topic_name):
+    """Lists all subscriptions for a given topic."""
+    pubsub_client = pubsub.Client()
+    topic = pubsub_client.topic(topic_name)
+
+    subscriptions = []
+    next_page_token = None
+    while True:
+        page, next_page_token = topic.list_subscriptions()
+        subscriptions.extend(page)
+        if not next_page_token:
+            break
+
+    for subscription in subscriptions:
+        print(subscription.name)
+
+
+def create_subscription(topic_name, subscription_name):
+    """Create a new pull subscription on the given topic."""
+    pubsub_client = pubsub.Client()
+    topic = pubsub_client.topic(topic_name)
+
+    subscription = topic.subscription(subscription_name)
+    subscription.create()
+
+    print('Subscription {} created on topic {}.'.format(
+        subscription.name, topic.name))
+
+
+def delete_subscription(topic_name, subscription_name):
+    """Deletes an existing Pub/Sub topic."""
+    pubsub_client = pubsub.Client()
+    topic = pubsub_client.topic(topic_name)
+    subscription = topic.subscription(subscription_name)
+
+    subscription.delete()
+
+    print('Subscription {} deleted on topic {}.'.format(
+        subscription.name, topic.name))
+
+
+def receive_message(topic_name, subscription_name):
+    """Receives a message from a pull subscription."""
+    pubsub_client = pubsub.Client()
+    topic = pubsub_client.topic(topic_name)
+    subscription = topic.subscription(subscription_name)
+
+    # Change return_immediately=False to block until messages are
+    # received.
+    results = subscription.pull(return_immediately=True)
+
+    print('Received {} messages.'.format(len(results)))
+
+    for ack_id, message in results:
+        print('* {}: {}, {}'.format(
+            message.message_id, message.data, message.attributes))
+
```

Comment:
yep, seems reasonable as well, just pointing out in unlikely case you overlooked.